libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/SetHttpsConnectionPropertiesEntitlement.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+/**
+ * An Entitlement to allow setting properties to a single Https connection after this has been created
```

Comment:
A thought: would it be a good practice to include in the javadocs a description of how each entitlement corresponds to existing `SecurityManager` permissions?

We'll want to record that somewhere, partly for our own reference over the next couple of months, and partly for plugin maintainers to port over their own policies. I suppose the entitlement class javadocs might not be the right home. 🤔